### ISSUE TITLE
[JS] Improve executable download troubleshooting in `@sentry/cli`

### ISSUE THREAD

**Issue by lobsterkatie** [getsentry/sentry-cli#1348]
Problem:

The post-install script in `@sentry/cli`, which attempts to download the version of the `sentry-cli` executable appropriate to the current platform, can error out for any number of reasons, and we don't give users a lot of help when it does.

Possible solutions:

- In error message, point users to the Node installation instructions at https://github.com/getsentry/sentry-cli#node
- Improve those installation instructions to explain how people are supposed to know what to call the binary if they're using the custom CDN option. (We say `The downloader will append "/<version>/sentry-cli-<dist>"`, but don't tell people how to determine `version` and `dist`.)

**Comment by kamilogorek** [comment]
> Improve those installation instructions to explain how people are supposed to know what to call the binary if they're using the custom CDN option. (We say The downloader will append "/<version>/sentry-cli-<dist>", but don't tell people how to determine version and dist.)

We already print a full url, including `version` and `dist` that they tried to install 🤔 

![image](https://user-images.githubusercontent.com/1523305/194024642-b737af58-b201-4690-bf8a-653b33d87762.png)

**Comment by lobsterkatie** [comment]
Yup, I've seen the error message, but that tells them the name they should use in that particular instance. I was thinking something more along the lines of https://github.com/getsentry/sentry-cli/pull/1350. (I ended up halfway doing it in the process of trying to explain what I meant in this comment, so I just went ahead and finished.)
